src/System.Text.Json/tests/JsonReaderTests.cs

```diff
@@ -230,11 +238,23 @@ private static Json ReadJson(string jsonString)
             var jsonReader = new JsonReader(jsonString);
             var jsonObjectMain = new Object();
             var jsonMembersMain = new List<Members>();
+            var jsonArrayMain = new Array();
+            var jsonElementsMain = new List<Elements>();
+
+            if (jsonString.Trim().Substring(0, 1) == "[")
```

Comment:
This is only in the JsonReaderTest code and not in the actual JsonReader. I am not sure if we the performance matters to this scale there.
Also, will StartsWith(Utf8CodeUnit) skip white space?
For instance, will "   x123".StartsWith((Utf8CodeUnit)(byte)'x') return true?